### PR TITLE
Support for querying the reflect reporting API and projects endpoints

### PR DESCRIPTION
**Pull request by cp**
- [x] Setup eslint and circleci
- [x] Add reporting namespace
- [x] Add projects namespace
- [x] Update README with documentation
- [x] Add jsdoc

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,93 @@
...
+
+Our reporting APIs are allow you to query the connections you've previously
+configured in Reflect.
+
+We currently only support the `report()` method. This method accepts four
+arguments:
+
+Name | Type | Description
+-----|------|--------------
+`connectionSlug` | string | The connection slug you wish to query
```
Comment (bradhe) [inline README.md]: This isn't the connection you want to query, but rather the data model (project) you want to query.

File: README.md [hunk 2]
```diff
@@ -1,24 +1,93 @@
...
+Note that it is required to specify either `dimensions` or `metrics`.
+
+This method returns a promise.
+
+Example:
+
+```javascript
+client.reporting.report('my-connection', ['Dimension 1'], ['Metric 1'])
+  .then((data) => {
+    console.log('data');
```
Comment (bradhe) [inline README.md]: quotes here.

File: src/client/reporting.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+function Reporting(client) {
+  this._client = client;
+}
+
+Reporting.prototype.report = function report(connectionSlug, dimensions, metrics, options = {}) {
```
Comment (bradhe) [inline src/client/reporting.js]: also project/data model instead of connection slug.

**Review by bradhe — CHANGES_REQUESTED** [review]
Minor thing

**Review by woodburndesigns — APPROVED** [review]

**Pull request merged**